### ISSUE TITLE
`generate/streamObject` send `undefined` schema when using `no-schema` for OpenAI models with structured outputs

### ISSUE THREAD

**Issue by elikor-figma** [vercel/ai#3436]
### Description

When making a `generateObject` or `streamObject` call with `output: 'no-schema'` and `structuredOuputs: true` on the model settings, the `OpenAIChatLanguageModel` implementation will send a request to OpenAI with the `response_format.json_schema.schema` being undefined. This seems to be due to the fact that the `jsonSchema` returned from the `noSchemaOutputStrategry` is `undefined` (see [here](https://github.com/vercel/ai/blob/main/packages/ai/core/generate-object/output-strategy.ts#L47)) which seems correct, but then is passed into `doGenerate` as `mode.schema`, which is ultimately used [here](https://github.com/vercel/ai/blob/main/packages/openai/src/openai-chat-language-model.ts#L218).
Passing an `undefined` schema always results in OpenAI responding with a 400, so this execution path will always fail. I believe that this can be resolved by adding a `&& mode.schema !== undefined` to [this line](https://github.com/vercel/ai/blob/main/packages/openai/src/openai-chat-language-model.ts#L214), where it seems logical to revert back to using `json_object` should no schema be provided.

### Code example

The example is a little silly because we are defining the model in-line which makes the `no-schema` and `structuredOutputs` stick out, however in the case where the code is a bit more complex and the model might be shared across multiple different types of requests this issues arises a bit more organically.

```
import { openai } from '@ai-sdk/openai';
import { generateObject } from 'ai';

async function main() {
  try {
    const result = await generateObject({
      model: openai('gpt-4o-2024-08-06', { structuredOutputs: true }),
      output: 'no-schema',
      prompt: 'Generate a lasagna recipe.',
    });
  } catch (e: unknown) {
    console.error("No Schema Error:", e)
  }
}
```

**Comment by lgrammel** [comment]
Great bug report! https://github.com/vercel/ai/pull/3445

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
The issue identifies a concrete failure in existing structured-output behavior: an undefined schema is sent to OpenAI and causes every affected request to fail.

### Evidence
- Using `generateObject` or `streamObject` with `output: 'no-schema'` and OpenAI `structuredOutputs: true` sends an undefined `response_format.json_schema.schema`.
- OpenAI rejects the resulting request with HTTP 400, making this execution path consistently fail.
- A maintainer explicitly called it a “bug report” and linked fix PR #3445.
